### PR TITLE
Increase the memory allocation for voronoi polygon coordinates

### PR DESCRIPTION
**Pull request by PaulWessel**
See #6306 for background.  It is hard to follow what I wrote some years ago regarding building these polygons, but hopefully this change will prevent us from running out of memory - after all these are small polygons.  After all, it is not entirely reassuring when I wrote the comment

`/* Need temp array to hold the coordinates of a single polygon [I hope this is large enough - don't know yet] */`

With this the example runs.  I also added an assert statement so that we are notified if this is still not enough.

### CODE REVIEW

**Review by joa-quim — COMMENTED** [review]
This change is not enough. The crash I got with this issue's example happened with this change that I already tried.

**Comment by PaulWessel** [comment]
Could you then please try 2* n_int_edges?  I mean the most complicated polygon cannot have too many vertices...

**Comment by joa-quim** [comment]
With the *2 it works fine. But one strange thing is that when I opened the issue the most common cause of error was that it hanged with the processor at 100%, not a crash. Crashes are easy to spot, those hangings not. But yesterday and today I was not able to make it happen again.

**Comment by PaulWessel** [comment]
Hm, yes I had to let it run a while in debug before it got to a stop point (yesterday).  But it was just slow in debug - it did get there.  But it is possible that the memory runover could have caused other issues related to the hanging.  let me know if you are OK to approved this - I rather not try to understand this part of the code.

**Review by joa-quim — APPROVED** [review]

**Comment by joa-quim** [comment]
I'll make a (richer) Matlab copy of triplot for you to play with.

File: src/gmt_support.c [hunk 1]
```diff
@@ -3800,8 +3800,8 @@ GMT_LOCAL struct GMT_DATASET * gmtsupport_voronoi_shewchuk (struct GMT_CTRL *GMT
 			if (i >= n_int_edges) edge_use[i] = 1;	/* These edges will only be used once (sign is not important here) */
 		}
 		/* Need temp array to hold the coordinates of a single polygon [I hope this is large enough - don't know yet] */
-		xcoord = gmt_M_memory (GMT, NULL, n_int_edges/2, double);
-		ycoord = gmt_M_memory (GMT, NULL, n_int_edges/2, double);
+		xcoord = gmt_M_memory (GMT, NULL, n_int_edges, double);
```
Comment (meghanrjones) [inline src/gmt_support.c:3803]: Doesn't this need to be 2*n_int_edges or some other value >n_int_edges for the crash to be resolved?

**Comment by PaulWessel** [comment]
Oops, sorry I thought @joa-quim had added 2 but I guess not committed.  OK will do this sneakily in master.

**Comment by PaulWessel** [comment]
Done
